src/test/java/org/folio/moduserstest/ProxyRelationshipsIT.java

```diff
@@ -0,0 +1,278 @@
+      ProxyRelationship.builder()
+        .userId(userId)
+        .proxyUserId(proxyUserId)
+        .build());
+
+    final var errors = proxiesClient.attemptToCreateProxyRelationship(
+        ProxyRelationship.builder()
+          .userId(userId)
+          .proxyUserId(proxyUserId)
+          .build())
+      .statusCode(is(422))
+      .extract().as(ValidationErrors.class);
+
+    assertThat(errors.getErrors().get(0).getMessage(),
+      is("Proxy relationship already exists"));
+  }
+
+  @Test
+  void canUpdateProxyRelationship() {
+    final var createdRelationship = proxiesClient.createProxyRelationship(
+      ProxyRelationship.builder()
+        .userId(generateId())
+        .proxyUserId(generateId())
+        .build());
+
+    final var changedProxyUserId = generateId();
+
+    proxiesClient.attemptToUpdateProxyRelationship(ProxyRelationship.builder()
+      .id(createdRelationship.getId())
+      .userId(generateId())
+      .proxyUserId(changedProxyUserId)
+      .build())
+      .statusCode(is(HTTP_NO_CONTENT));
+
+    final var fetchedRelationship = proxiesClient.getProxyRelationship(
+      createdRelationship.getId());
+
+    assertThat(fetchedRelationship.getProxyUserId(), is(changedProxyUserId));
+  }
+
+  @Test
+  void cannotUpdateUnknownProxyRelationship() {
+    // Create a relationship to ensure it isn't accidentally deleted
+    proxiesClient.createProxyRelationship(
+      ProxyRelationship.builder()
+        .userId(generateId())
+        .proxyUserId(generateId())
+        .build());
+
+    proxiesClient.attemptToUpdateProxyRelationship(ProxyRelationship.builder()
+        .id(generateId())
+        .userId(generateId())
+        .proxyUserId(generateId())
+        .build())
+      .statusCode(is(HTTP_NOT_FOUND));
+  }
+
+  @Test
+  void canFindProxiesForUser() {
+    final var userId = generateId();
+    final var differentUserId = generateId();
+    final var firstProxyUserId = generateId();
+    final var secondProxyUserId = generateId();
+
+    proxiesClient.createProxyRelationship(
+      ProxyRelationship.builder()
+        .userId(userId)
+        .proxyUserId(firstProxyUserId)
+        .build());
+
+    proxiesClient.createProxyRelationship(
+        ProxyRelationship.builder()
+          .userId(userId)
+          .proxyUserId(secondProxyUserId)
+          .build());
+
+    proxiesClient.createProxyRelationship(
+      ProxyRelationship.builder()
+        .userId(differentUserId)
+        .proxyUserId(secondProxyUserId)
+        .build());
+
+    final var proxyRelationships = proxiesClient.getProxyRelationships(
+      String.format("userId==%s", userId));
+
+    assertThat(proxyRelationships.getTotalRecords(), is(2));
+
+    final var allProxies = proxyRelationships.getProxiesFor()
+      .stream()
+      .map(ProxyRelationship::getProxyUserId)
+      .collect(Collectors.toList());
+
+    assertThat(allProxies, containsInAnyOrder(firstProxyUserId, secondProxyUserId));
```

Comment:
A more rest-assured-like test:
```
    proxiesClient.getProxyRelationships(String.format("userId==%s", userId),
        "totalRecords", is(2),
        "proxiesFor.collect { p -> p.proxyUserId }", containsInAnyOrder(firstProxyUserId, secondProxyUserId));
```
Requires this method:
```
  public ProxyRelationships getProxyRelationships(String cqlQuery,
      String path, Matcher<?> matcher, Object... pathMatcherPairs) {
    return given()
      .config(config)
      .spec(requestSpecification)
      .when()
      .queryParam("query", cqlQuery)
      .get("/proxiesfor")
      .then()
      .statusCode(HTTP_OK)
      .body(path, matcher, pathMatcherPairs)
      .extract().as(ProxyRelationships.class);
  }
```

Reply:
I think we have different design considerations around the use of rest assured to provide readable tests.

I have no doubt that my rushed conversion of the existing tests means I've made compromises where this could be improved.

This is certainly one of those cases. I was tempted to extract a method into the ProxyRelationships class to hide the stream manipulation.

I think I'm struggling with much of this feedback because I don't understand why relying on JSON path based assertions in RestAssured is better.

I think this is especially the case when the quantity of assertions grows, passing in a map or list of assertions increases the chances that they will get lost in the details and be harder to understand.

To me, checking the status code in the chain of methods from RestAssured is already a compromise as it starts to mix together two separate concerns (making the request and checking the response).